### ISSUE TITLE
[bug]: Focus outline crossing borders of list items

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#7455]
### 🐞 Describe the bug

<img width="1243" height="695" alt="Image" src="https://github.com/user-attachments/assets/ee7ba755-0f82-4fb2-b036-f67bffc0fcbc" />

https://career.post.ch/de/lernende

Focus outline should be inset in link lists so that the focus outline does not interfere with the borders of the items.

**Comment by yomonom** [comment]
This also applies to accordions:

<img width="1169" height="223" alt="Image" src="https://github.com/user-attachments/assets/393894fa-25e5-411a-b580-a726715b90cb" />
